lib/rspec-puppet/adapters.rb

```diff
@@ -146,6 +146,9 @@ def settings_map
 
       def catalog(node, exported)
         node.environment = current_environment
+        # Override $::environment to workaround PUP-5835, where Puppet otherwise
+        # stores a symbol for the parameter
+        node.parameters['environment'] = current_environment.name.to_s if Puppet.version.to_f >= 4.0 && Puppet.version.to_f < 4.4
```

Comment:
This might be more robust (version independent) by saying:

```
node.parameters['environment'] = current_environment.name.to_s if node.parameters['environment'] != node.parameters['environment'].to_s
```